v7.c

```diff
@@ -10886,37 +10885,57 @@ int slre_replace(struct slre_loot *loot, const char *src, const char *rstr,
   return size;
 }
 
-#ifdef SLRE_TEST
+int slre_match(const char *re, unsigned flags, const char *str,
+               struct slre_loot *loot) {
+  struct slre_prog *prog = NULL;
+  int res;
 
-static int re_replace(struct slre_loot *loot, const char *src, const char *rstr,
-                      char **dst) {
-  struct slre_loot newsub;
-  struct slre_cap *t = newsub.caps;
-  char *d;
-  int osz = slre_replace(loot, src, rstr, &newsub);
-  int i = newsub.num_captures;
-  if (osz < 0) {
-    printf("slre_replace return: '%s'\n", SLRE_osz);
-    return 0;
+  if ((res = slre_compile(re, &prog)) == SLRE_OK) {
+    res = slre_exec(prog, flags, str, loot);
+    slre_free(prog);
   }
-  *dst = NULL;
-  if (osz) *dst = SLRE_MALLOC(osz + 1);
-  if (!*dst) return 0;
-  d = *dst;
-  do {
-    size_t len = t->end - t->start;
-    memcpy(d, t->start, len);
-    d += len;
-    t++;
-  } while (--i);
-  *d = '\0';
-  return osz;
+
+  return res;
+}
+
+
+#ifdef SLRE_TEST
```

Comment:
I commented on the makefile because it was relevant there, but it might as well be relevant here :-)

Should we have a makefile rule that actually runs this code or do you plan to put it in the slre repo?

For lazymen like me who don't want to type `gcc -o /tmp/slre src/slre.c src/utf.c -DSLRE_TEST` :-)